scalafix/rules/src/main/scala/migrate/ExplicitImplicitsRule.scala

```diff
@@ -76,23 +80,34 @@ class ExplicitImplicitsRule[G <: Global](g: G) {
 
   }
 
-  private def collectImplicit(globalTree: g.Tree, original: Tree): Option[List[g.Symbol]] = {
+  private def collectImplicit(globalTree: g.Tree, original: Tree): Option[(List[g.Symbol], Boolean)] = {
     val collectedTree: Seq[g.Tree] =
       globalTree.collect {
         case t if CompilerService.equalForPositions(t.pos, original.pos) => t
       }.filter(_.isInstanceOf[g.ApplyToImplicitArgs])
 
-    collectedTree.head.asInstanceOf[g.Apply].fun.isTerm
     collectedTree.collectFirst {
       // at the same position we are supposed to have maximum one ApplyToImplicitArgs
       // except it there is also an implicit conversion that takes implicits.
       // See Mix.scala example
       case g.Apply(fun, args) if !fun.isInstanceOf[g.ApplyImplicitView] => {
         val listOfArgs = args.map(_.symbol.asInstanceOf[g.Symbol])
-        listOfArgs
+        (listOfArgs, isEtaExpansion(fun, original))
       }
     }
 
   }
 
+  private def toAddImplicit(originalTree: Tree, args: List[String], isEtaExpansion: Boolean): Patch =
+    if (isEtaExpansion) Patch.addRight(originalTree, "(_)" + "(" + args.mkString(", ") + ")")
```

Comment:
What if it is eta-expanded with more than one argument? Maybe you can have the size of the first argument list. Or you can try the following but I am not sure it will work:
```scala
if (isEtaExpansion) Patch.addRight("(" + originalTree +  "_ )" + "(" + args.mkString(", ") + ")")
```